src/pages/task/ActionButtons.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useRef, useState, useEffect } from "react";
+import { Button } from "components/Button";
+import styled from "@emotion/styled";
+import { useOnClickOutside } from "hooks";
+import { InputNumber, Popconfirm } from "antd";
+import get from "lodash/get";
+import { Body } from "@leafygreen-ui/typography";
+import { useParams } from "react-router-dom";
+import { useMutation } from "@apollo/react-hooks";
+import { ABORT_TASK } from "gql/mutations/abort-task";
+import { RESTART_TASK } from "gql/mutations/restart-task";
+import { SCHEDULE_TASK } from "gql/mutations/schedule-task";
+import { UNSCHEDULE_TASK } from "gql/mutations/unschedule-task";
+import { SET_TASK_PRIORTY } from "gql/mutations/set-task-priority";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  SetTaskPriorityMutation,
+  SetTaskPriorityMutationVariables,
+  AbortTaskMutation,
+  AbortTaskMutationVariables,
+  RestartTaskMutation,
+  RestartTaskMutationVariables,
+  ScheduleTaskMutation,
+  ScheduleTaskMutationVariables,
+  UnscheduleTaskMutation,
+  UnscheduleTaskMutationVariables,
+} from "gql/generated/types";
+import { PageButtonRow } from "components/styles";
+import { DropdownItem, ButtonDropdown } from "components/ButtonDropdown";
+
+interface Props {
+  priority?: number;
+  canAbort: boolean;
+  canRestart: boolean;
+  canSchedule: boolean;
+  canUnschedule: boolean;
+  canSetPriority: boolean;
+}
+
+export const ActionButtons = ({
+  canAbort,
+  canRestart,
+  canSchedule,
+  canSetPriority,
+  canUnschedule,
+  ...props
+}: Props) => {
+  const { success, error } = useBannerDispatchContext();
+  const wrapperRef = useRef(null);
+  const priorityRef = useRef(null);
+  const [isVisible, setIsVisible] = useState(false);
+  const [priority, setPriority] = useState<number>(props.priority);
+  const { id: taskId } = useParams<{ id: string }>();
+
+  const [scheduleTask, { loading: loadingScheduleTask }] = useMutation<
+    ScheduleTaskMutation,
+    ScheduleTaskMutationVariables
+  >(SCHEDULE_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task marked as scheduled");
+    },
+    onError: (err) => {
+      error(`Error scheduling task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [unscheduleTask, { loading: loadingUnscheduleTask }] = useMutation<
+    UnscheduleTaskMutation,
+    UnscheduleTaskMutationVariables
+  >(UNSCHEDULE_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task marked as unscheduled");
+    },
+    onError: (err) => {
```

Comment:
super minor nit, sorry :(. Lets decide on how errors are passed/named you passed just an `e` earlier in this pr. For consistencies sake i think it would be nice if we had a common pattern.

Reply:
what's the value add here?

Reply:
I think it was more for consistency since the error was passed in with a different variable name earlier in the pr but its not to important.